src/components/core/common-behaviors/icon-name-mixin.ts

```diff
@@ -0,0 +1,24 @@
+import type { LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+
+import type { Constructor } from './constructor';
+
+export declare class SbbIconNameInterface {
+  public iconName: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const SbbIconNameMixin = <T extends Constructor<LitElement>>(
+  superClass: T,
+): Constructor<SbbIconNameInterface> & T => {
+  class SbbIconName extends superClass implements Partial<SbbIconNameInterface> {
+    /**
+     * The icon name we want to use, choose from the small icon variants
+     * from the ui-icons category from here
+     * https://icons.app.sbb.ch.
+     */
+    @property({ attribute: 'icon-name', reflect: true }) public iconName?: string;
```

Comment:
Done in [8b9d0f9](https://github.com/lyne-design-system/lyne-components/pull/2364/commits/8b9d0f94fa4275f797fff1aa24f2502eabc88d7e)
the icon-name mixin has its basic method and each common mixin wraps it with its logic (basically it's the same, with a different css class, but it's open to change if needed).
What do you think?

Reply:
Could not follow the commit (broken link) and did not fully understand your statement. But the render method you introduced is what I meant, so it looks good so far. what do you mean with css class?

Reply:
Rebase changed commit hashes :'(
however, now the icon-name mixin has a `renderSlotIcon` method which renders the slot; any `<component>-common-mixin` has its own `renderSlotIcon` which wraps the said method with a span and a css class, e.g
[button-common at line 85](https://github.com/lyne-design-system/lyne-components/pull/2364/commits/b7b74513572499743f7d312cfa6a82dabd0fefc2#diff-b8b548d1d12faf82b9c8c6e365af9acd64b7ed0912c7edceb40db564a322928a) or [header-common at line 66](https://github.com/lyne-design-system/lyne-components/pull/2364/commits/b7b74513572499743f7d312cfa6a82dabd0fefc2#diff-0e2626d36febd91ae60d4be869e2e25518f9f876e401508a62ae6fb5fd0e1653) and so on..

Reply:
ah thanks for the link, I understand it now. so my question back is, is this extra step with override not kind of too complicated or could not just the renderSlotIcon() directly be called in the specific templates and having the span class wrapper there without the extra method override.

Reply:
Yeah, I have mixed feeling about this, I did it in this way because the span wrapper it's the same in both link and button case, but I can revert it if we feel it's an overkill

Reply:
It's ok, but maybe also ask @kyubisation ? :)

Reply:
I'm not sure I fully understand.
I will try to read the changes.
However, I already noticed that the methods are public. I would prefer them to be protected.